components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -383,38 +383,49 @@ void EthTxController::ApproveHardwareTransaction(
   }
 }
 
+void EthTxController::GetTransactionMessageToSign(
+    const std::string& tx_meta_id,
+    GetTransactionMessageToSignCallback callback) {
+  std::unique_ptr<EthTxStateManager::TxMeta> meta =
+      tx_state_manager_->GetTx(tx_meta_id);
+  if (!meta) {
+    LOG(ERROR) << "No transaction found";
+    std::move(callback).Run("");
+    return;
+  }
+  uint256_t chain_id = 0;
+  if (!HexValueToUint256(rpc_controller_->GetChainId(), &chain_id)) {
+    std::move(callback).Run("");
+    return;
+  }
+  auto message = meta->tx->GetMessageToSign(chain_id, false);
+  auto encoded = brave_wallet::ToHex(message);
+  std::move(callback).Run(encoded);
+}
+
 void EthTxController::OnGetNextNonceForHardware(
     std::unique_ptr<EthTxStateManager::TxMeta> meta,
-    ApproveHardwareTransactionCallback callback,
+    GetNonceForHardwareTransactionCallback callback,
     bool success,
     uint256_t nonce) {
   if (!success) {
     meta->status = mojom::TransactionStatus::Error;
     tx_state_manager_->AddOrUpdateTx(*meta);
     LOG(ERROR) << "GetNextNonce failed";
-    std::move(callback).Run(false, "");
+    std::move(callback).Run("");
```

Comment:
why did you remove the bool?

Reply:
it is result of refactoring to share interfaces between Trezor/Ledger, it returns `nonce` string now instead of old bool, I think we can change it to optional

Reply:
but the bool was to indicate success/failure. It previously returned a string and bool

Reply:
changed to use optional type